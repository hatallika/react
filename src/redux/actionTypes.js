
export const GET_POSTS = "GET_POSTS";
export const GET_POSTS_LOADING = "GET_POSTS_LOADING";
export const GET_POSTS_ERROR = "GET_POSTS_ERROR";

export const DELETE_CHAT = "deletechat";
export const ADD_CHAT = "addchat";

export const ADD_MESSAGE = 'addmessage';
export const DELETE_MESSAGE = 'delmessage';
export const ROBOT_ANSWER = 'robotanswer';