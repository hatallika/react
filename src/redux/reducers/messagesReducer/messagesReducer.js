const initialState = {
    messages: [
        {id: 1, author:"author1",message: "Hello!", chatId: 1},
        {id: 2, author:"author2",message: "Hello! My friends!", chatId: 1},
        {id: 3, author:"author3",message: "Hi! I am here!", chatId: 2},
        {id: 4, author:"author3",message: "Hi!", chatId: 3},
    ]
};

export const messagesReducer = (state = initialState, action) => {
    switch (action.type) {
        case('addmessage'): return {
            ...state,
            messages: [...state.messages, {
                id: action.payload.id,
                author: action.payload.author,
                message: action.payload.message,
                chatId: action.payload.chatId
            }]
        }
        case ('delmessage'): return {
            ...state,
            messages: state.messages.filter((message)=> message.id !== action.payload)
        }

        default: return state;
    }
}

