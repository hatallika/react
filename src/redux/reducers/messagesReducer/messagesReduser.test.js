import {messagesReducer} from "./messagesReducer";
import {ADD_MESSAGE} from "../../actionTypes";
const initialState = {
    messages: [
        {id: 1, author:"author1",message: "Hello!", chatId: 1},
        {id: 2, author:"author2",message: "Hello! My friends!", chatId: 1},
        {id: 3, author:"author3",message: "Hi! I am here!", chatId: 2},
        {id: 4, author:"author3",message: "Hi!", chatId: 3},
    ]
};

export const addMessages = (message) => ({
    type: ADD_MESSAGE,
    payload: {
        id: 1,
        author: 'author',
        message: message,
        chatId: 1,
    }

});
describe('messageReducer', () => {
    //проверка изменения стейта после работы редюсера.
    test('messageReducerState', () => {
        const messageStore = messagesReducer(initialState, addMessages('message'));
        expect(messageStore).toEqual({
            messages: [
                {id: 1, author:"author1",message: "Hello!", chatId: 1},
                {id: 2, author:"author2",message: "Hello! My friends!", chatId: 1},
                {id: 3, author:"author3",message: "Hi! I am here!", chatId: 2},
                {id: 4, author:"author3",message: "Hi!", chatId: 3},
                {id: 1, author:"author",message: 'message', chatId: 1},
            ]})
        //expect(messageStore).toMatchSnapshot();
    })
    //делаем снимок состояния, он выдаст ошибку при изменении редюсера
    //обновить снэпшот можно с помощью флага -u при запуске тестов в терминале
    test('snapshot', () => {
        const messageStore = messagesReducer(initialState, addMessages('message'));
        expect(messageStore).toMatchSnapshot();
    })
} )
