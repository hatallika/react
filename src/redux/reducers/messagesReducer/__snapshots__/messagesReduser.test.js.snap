// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`messageReducer 1`] = `
Object {
  "messages": Array [
    Object {
      "author": "author1",
      "chatId": 1,
      "id": 1,
      "message": "Hello!",
    },
    Object {
      "author": "author2",
      "chatId": 1,
      "id": 2,
      "message": "Hello! My friends!",
    },
    Object {
      "author": "author3",
      "chatId": 2,
      "id": 3,
      "message": "Hi! I am here!",
    },
    Object {
      "author": "author3",
      "chatId": 3,
      "id": 4,
      "message": "Hi!",
    },
  ],
}
`;

exports[`messageReducer snapshot 1`] = `
Object {
  "messages": Array [
    Object {
      "author": "author1",
      "chatId": 1,
      "id": 1,
      "message": "Hello!",
    },
    Object {
      "author": "author2",
      "chatId": 1,
      "id": 2,
      "message": "Hello! My friends!",
    },
    Object {
      "author": "author3",
      "chatId": 2,
      "id": 3,
      "message": "Hi! I am here!",
    },
    Object {
      "author": "author3",
      "chatId": 3,
      "id": 4,
      "message": "Hi!",
    },
    Object {
      "author": "author",
      "chatId": 1,
      "id": 1,
      "message": "message",
    },
  ],
}
`;
